core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -259,99 +250,113 @@ oppia.controller('StateResponses', [
+                params: {
+                  state_name: _stateName
+                }
+              }).success(function(response) {
+                $scope.showUnresolvedAnswers(response.unresolved_answers);
+              }).error(function(response) {
+                $scope.showUnresolvedAnswers([]);
+              });
             };
 
-            FocusManagerService.setFocus('testInteractionInput');
+            $scope.showUnresolvedAnswers = function(unresolvedAnswers) {
+              $scope.showLoadingDots = false;
+              $scope.unresolvedAnswers = [];
+
+              unresolvedAnswers.forEach(function(item) {
+                var answer = item.answer;
+                var answerTemplate = (
+                  ExplorationHtmlFormatterService.getAnswerHtml(
+                    answer, stateInteractionIdService.savedMemento,
+                    stateCustomizationArgsService.savedMemento));
+                var classificationResult = (
+                  AnswerClassificationService.getMatchingClassificationResult(
+                    _explorationId, _stateName, _state, answer, rulesService));
+                var feedbackHtml = (
+                  classificationResult.outcome.feedback.getHtml());
+                var classificationType = (
+                  classificationResult.classificationCategorization);
+                if (classificationType !== EXPLICIT_CLASSIFICATION &&
+                  classificationType !== TRAINING_DATA_CLASSIFICATION &&
+                  !TrainingDataService.isConfirmedUnclassifiedAnswer(answer)) {
+                  $scope.unresolvedAnswers.push({
+                    answer: answer,
+                    answerTemplate: answerTemplate,
+                    classificationResult: classificationResult,
+                    feedbackHtml: feedbackHtml
+                  });
+                }
+              });
+            };
+
+            $scope.confirmToTrainingData = function(answerIndex) {
+              answer = $scope.unresolvedAnswers[answerIndex];
+              $scope.unresolvedAnswers.splice(answerIndex, 1);
+              if (answer.classificationResult.classificationCategorization ===
+                  DEFAULT_OUTCOME_CLASSIFICATION) {
+                TrainingDataService.trainDefaultResponse(answer.answer);
+                return;
+              }
+
+              TrainingDataService.trainAnswerGroup(
+                answer.classificationResult.answerGroupIndex, answer.answer);
+            };
+
+            $scope.finishTrainingAnswer = function() {
+              $scope.unresolvedAnswers.splice($scope.selectedAnswerIndex, 1);
+            };
+
+            $scope.openTrainUnresolvedAnswerModal = function(answerIndex) {
+              $scope.selectedAnswerIndex = answerIndex;
+              answer = $scope.unresolvedAnswers[answerIndex].answer;
+              return TrainingModalService.openTrainUnresolvedAnswerModal(
+                answer, true, $scope.finishTrainingAnswer);
```

Comment:
Is it possible to dismiss the modal without training the answer? If so what happens to $scope.selectedAnswerIndex?

Reply:
But that would require adding another callback. It does no harm even if we keep it as it is.

Reply:
Ah -- I was going to object to that, because it's conceptually muddy. If no answer is selected then we shouldn't have $scope.selectedAnswerIndex indicating otherwise.

However, I just looked at this more carefully, and I don't see the point of this being declared on $scope in the first place. If you put $scope.finishTrainingAnswer inline, then I think you can just have `var selectedAnswerIndex` and then that variable just becomes internal to this function. That seems like it would be a lot clearer.

Reply:
Ok, will do.